### PR TITLE
Example of schema validation

### PR DESCRIPTION
**Pull request by stephenrichards**
## What

[Link to story](https://dsdmoj.atlassian.net/browse/AP-XXX)

Describe what you did and why.

## Checklist

Before you ask people to review this PR:

- Tests and rubocop should be passing: `bundle exec rake`
- Github should not be reporting conflicts; you should have recently run `git rebase main`.
- There should be no unnecessary whitespace changes. These make diffs harder to read and conflicts more likely.
- The PR description should say what you changed and why, with a link to the JIRA story.
- You should have looked at the diff against main and ensured that nothing unexpected is included in your changes.
- You should have checked that the commit messages say why the change was made.

### CODE REVIEW

File: spec/json_spike/json_validator_spec.rb [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    {
+      '$id' => "check_financial_eligibility/schemas/applicant",
+      type: :object,
+      required: [:applicant],
+      properties: {
+        applicant: {
+          type: :object,
+          properties: {
+            date_of_birth: {
+              type: :string,
```
Comment (jsugarman) [inline spec/json_spike/json_validator_spec.rb:30]:
```
"date": {
  "type": "string",
  "format": "date"
}
```

File: spec/json_spike/json_validator_spec.rb [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+              pattern: "^[1-2][0-9]{3}-[0-1][0-9]-[0-3][0-9]$"
+            },
+            involvement_type: {
+              type: :string,
+              pattern: "^applicant$"
```
Comment (jsugarman) [inline spec/json_spike/json_validator_spec.rb:35]: can use enum:

```
involvement_type : {
          type : :string,
          enum: ["applicant"]
        },
```

File: spec/json_spike/json_validator_spec.rb [hunk 3]
```diff
@@ -0,0 +1,93 @@
+require 'rspec'
+
+describe 'JsonValidator' do
```
Comment (jsugarman) [inline spec/json_spike/json_validator_spec.rb:3]:
```suggestion
RSpec.describe 'JsonValidator' do
```

File: spec/json_spike/json_validator_spec.rb [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+              pattern: "^[1-2][0-9]{3}-[0-1][0-9]-[0-3][0-9]$"
+            },
+            involvement_type: {
+              type: :string,
+              pattern: "^applicant$"
+            },
+            has_partner_opponent: {type: :boolean },
+            receives_qualifying_benefit: {
+              type: :string,
+              pattern: "(^yes$)|(^no$)"
```
Comment (jsugarman) [inline spec/json_spike/json_validator_spec.rb:40]:
```suggestion
              enum: ["yes", "no"]
```

or 
```suggestion
     type: :boolean,
```

**Review by jsugarman — COMMENTED** [review]
added some possible tweaks

File: spec/json_spike/json_validator_spec.rb (hunk 1 above)
Comment (stephenrichards) [reply spec/json_spike/json_validator_spec.rb:30]: I can't get this to work - it considers anything valid
```
>>>>>>>>>  /Users/stephenrichards/moj/check/spec/json_spike/applicant_schema_spec.rb:39 <<<<<<<<<<
{"$schema"=>"http://json-schema.org/draft-04/schema", "$id"=>"check_financial_eligibility/schemas/applicant", "type"=>"object", "required"=>["applicant"], "properties"=>{"applicant"=>{"type"=>"object", "properties"=>{"date_of_birth"=>{"type"=>"string", "format"=>"date"}, "involvement_type"=>{"enum"=>["applicant"]}, "has_partner_opponent"=>{"type"=>"boolean"}, "receives_qualifying_benefit"=>{"enum"=>["yes", "no"]}}, "required"=>["date_of_birth", "involvement_type", "has_partner_opponent", "receives_qualifying_benefit"]}}}
>>>>>>>>>  /Users/stephenrichards/moj/check/spec/json_spike/applicant_schema_spec.rb:41 <<<<<<<<<<
{"applicant":{"date_of_birth":"3xx0024-1223","involvement_type":"defendant","has_partner_opponent":false,"receives_qualifying_benefit":"yes"}}
[
    [0] "The property '#/applicant/involvement_type' value \"defendant\" did not match one of the following values: applicant in schema 80f83d28-71d8-594f-ade3-747223b5d850"
]```
no error message for the dodgy date

**Comment by colinbruce** [comment]
Closed - work undertaken in #927

**Pull request closed without merging**